lib/api/core/hotelClerk.js

```diff
@@ -410,66 +336,39 @@ function removeListRoomsInCollection (index, collection, rooms) {
 function createRoom (index, collection, filters) {
   var
     deferred = q.defer(),
-    self = this,
-    sortedFilters = [],
-    stringifiedFilters,
-    stringifiedObject,
     roomId;
 
-  /*
-   Make sure an index and a collection are provided
-   */
   if (!index || !collection) {
     deferred.reject(new BadRequestError('Cannot subscribe without an index and a collection'));
     return deferred.promise;
   }
 
-  /*
-   Ensure the resulting room id is identical even if the order of filters terms is changed.
-   This should do the trick for now, but a deep sort() would probably be better.
-    */
-  if (!filters) {
-    filters = {};
-  }
-
-  Object.keys(filters).forEach(key => sortedFilters.push([key, filters[key]]));
-  stringifiedFilters = stringify(sortedFilters.sort((a, b) => {
-    return String(a[0]).localeCompare(b[0]);
-  }));
-
-  stringifiedObject = stringify({index: index, collection: collection, filters: stringifiedFilters});
+  roomId = this.kuzzle.dsl.createFilterId(index, collection, filters);
```

Comment:
This code has been removed because it's redundant with the `json-stable-stringify` module.

I'm ashamed to say that it was me who have added these transformations a few months ago without reading `json-stable-stringify` documentation. This module does exactly what we did here: sort the JSON terms to ensure we'll get the same hash when encoding differently sorted identical filters.